axelor-talent/src/main/resources/views/JobPosition.xml

```diff
@@ -70,4 +80,24 @@
 		<context name="_responsibleId" expr="eval:employee.id"/>
 	</action-view>
 	
+	<action-record name="action-job-position-record-status-cancel" model="com.axelor.apps.talent.db.JobPosition">
+    	<field name="statusSelect" expr="eval: 4"/>
+    </action-record>
+    
+    <action-record name="action-job-position-record-status-open" model="com.axelor.apps.talent.db.JobPosition">
+    	<field name="statusSelect" expr="eval: 1" if="statusSelect == 0"/>
+    	<field name="statusSelect" expr="eval: 2" if="statusSelect == 1"/>
+    	<field name="statusSelect" expr="eval: 3" if="statusSelect == 2"/>
+    	<field name="statusSelect" expr="eval: 0" if="statusSelect == 4"/>
+    </action-record>
+    
+    <action-attrs name="action-job-position-attrs-change-status-button-title">
```

Comment:
Don't create single button for all status changes. Use separate buttons for each and display according to status.